samples/gibbs_ensemble/client_system.py

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright (C) 2013-2019 The ESPResSo project
+#
+# This file is part of ESPResSo.
+#
+# ESPResSo is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ESPResSo is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+This script uses the System class from the :file:`gibbs_ensemble.py`
+script and creates a subclass in which you can create your system, add
+energy corrections etc.
+"""
+
+import espressomd
+import gibbs_ensemble
+import numpy as np
+
+espressomd.assert_features("LENNARD_JONES")
+
+# Lennard Jones parameters
+LJ_EPSILON = 1.0
+LJ_SHIFT = 0.0
+LJ_SIGMA = 1.0
+LJ_CUTOFF = 2.5 * LJ_SIGMA
+
+
+def calc_tail_correction(density):
+    """ Lennard Jones truncation correction (See Frenkel, Smit, Understanding
+        molecular simulation, eq. (3.2.5)) """
+    return 8. / 3. * np.pi * density * LJ_EPSILON * LJ_SIGMA**3 * \
+        (1. / 3. * np.power(LJ_SIGMA / LJ_EPSILON, 9) -
+         np.power(LJ_SIGMA / LJ_EPSILON, 3))
+
+
+class Gibbs_Client(gibbs_ensemble.Client):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @property
+    def energy(self):
+        """ Additional correction terms """
+        return super().energy + calc_tail_correction(self.density)
+
+    def init_system(self):
+        """ Initialize the system (this is run only when the run method of the
+            process is run """
+        self.system = set_up_system(self.init_box_length, self.init_num_part)
+
+
+def set_up_system(box_length, num_part):
+    """ Use this function to create the system """
+
+    system = espressomd.System(box_l=3 * [box_length])
+
+    system.cell_system.set_n_square(use_verlet_lists=False)
+    system.cell_system.skin = 0
+    system.time_step = 0.01
+
+    for _ in range(num_part):
+        system.part.add(pos=np.random.random(3) * box_length)
```

Comment:
Please use a single call to system.part.add() with an (N,3)-shaped array of random numbers for positions. Avoids the loop.